src/lavinmq/exchange/headers.cr

```diff
@@ -6,53 +6,61 @@ module LavinMQ
       "headers"
     end
 
+    @bindings = Hash(AMQP::Table, Set(Destination)).new do |h, k|
+      h[k] = Set(Destination).new
+    end
+
+    def bindings_details : Iterator(BindingDetails)
+      @bindings.each.map do |args, ds|
+        ds.map do |d|
+          BindingDetails.new(name, vhost.name, "", args, d)
+        end
+      end.flatten
```

Comment:
Should be possible to use `Iterator#flat_map` to get rid of allocations. You can probably do `ds.each.map` too?